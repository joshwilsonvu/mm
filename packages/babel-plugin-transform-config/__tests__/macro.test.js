const pluginTester = require("babel-plugin-tester").default;
const macrosPlugin = require("babel-plugin-macros");

// replaces project path with <PROJECT_ROOT> in absolute paths
const jestSerializerPath = require("jest-serializer-path");
expect.addSnapshotSerializer(jestSerializerPath);

pluginTester({
  plugin: macrosPlugin,
  tests: {
    works: {
      code: `
        import transformConfig from "../../macro";

        export default transformConfig({
          modules: [
            {
              module: "a-module",
              position: "top_left"
            }
          ]
        })
      `,
      snapshot: true,
    },
    "accepts a simple options object": {
      code: `
      import transformConfig from "../../macro";

      export default transformConfig({
        modules: [
          {
            module: "the-module"
          }
        ]
      }, {
        modulesPath: "../../custom-modules"
      })
      `,
      snapshot: true,
    },
  },
  babelOptions: {
    filename: require.resolve("./config/dummy"),
    babelrc: false,
    configFile: false,
    root: __dirname,
  },
});
