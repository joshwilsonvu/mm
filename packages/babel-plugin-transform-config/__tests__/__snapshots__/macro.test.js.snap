// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`macros accepts a simple options object: accepts a simple options object 1`] = `

import transformConfig from "../../macro";

export default transformConfig({
  modules: [
    {
      module: "the-module"
    }
  ]
}, {
  modulesPath: "../../custom-modules"
})

      ↓ ↓ ↓ ↓ ↓ ↓

export default {
  modules: [
    {
      module: "the-module",
      _component: require("react").lazy(() =>
        import(
          /* webpackChunkName: "the-module" */

          /* webpackPreload: true */
          "<PROJECT_ROOT>/packages/babel-plugin-transform-config/__tests__/custom-modules/the-module/index.js"
        )
      ),
      _path:
        "<PROJECT_ROOT>/packages/babel-plugin-transform-config/__tests__/custom-modules/the-module/index.js",
    },
  ],
};


`;

exports[`macros works: works 1`] = `

import transformConfig from "../../macro";

export default transformConfig({
  modules: [
    {
      module: "a-module",
      position: "top_left"
    }
  ]
})

      ↓ ↓ ↓ ↓ ↓ ↓

export default {
  modules: [
    {
      module: "a-module",
      _component: require("react").lazy(() =>
        import(
          /* webpackChunkName: "a-module" */

          /* webpackPreload: true */
          "<PROJECT_ROOT>/packages/babel-plugin-transform-config/__tests__/modules/a-module/a-module.js"
        )
      ),
      _path:
        "<PROJECT_ROOT>/packages/babel-plugin-transform-config/__tests__/modules/a-module/a-module.js",
      position: "top_left",
    },
  ],
};


`;
